modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/model/listener/DDMFormInstanceModelListener.java

```diff
@@ -66,12 +67,11 @@ public void onAfterRemove(DDMFormInstance ddmFormInstance)
 					getFormInstanceReportByFormInstanceId(
 						ddmFormInstance.getFormInstanceId());
 
-			if (ddmFormInstanceReport == null) {
-				return;
-			}
-
-			_ddmFormInstanceReportLocalService.deleteDDMFormInstanceReport(
-				ddmFormInstanceReport.getFormInstanceReportId());
+			_ddmFormInstanceReportPortalExecutor.execute(
+				() ->
+					_ddmFormInstanceReportLocalService.
+						deleteDDMFormInstanceReport(
+							ddmFormInstanceReport.getFormInstanceReportId()));
```

Comment:
@marcosapmf, isn't necessary do this same change on the method `onAfterCreate`?

Reply:
@natocesarrego
No, it is not. `onAfterCreate` just creates an empty report and there is no process there. Therefore, we can keep it as synchronous :)